cmake/ExternalProjects.cmake

```diff
@@ -20,8 +20,8 @@ FetchContent_Declare(wamr_ext
 )
 
 FetchContent_Declare(faabric_ext
-    GIT_REPOSITORY "https://github.com/faasm/faabric.git"
-    GIT_TAG "v0.0.12"
+    URL "https://github.com/faasm/faabric/archive/v0.0.13.tar.gz"
```

Comment:
I figured out that if our idea was to pin to a specific tag, we could as well just use the tag-specific `URL`, and avoid all the unexpected `git` behaviour like not fetching new tags/branches.

Additionally, by including the `URL_MD5` hash, we ensure the integrity of the content (not a real concern, but nice to have), as well as easily caching downloads. It is also [strongly recommended](https://cmake.org/cmake/help/latest/module/ExternalProject.html#module:ExternalProject) in the docs (look for `URL_HASH`).

To obtain such hash, I just use `md5sum` from the command line. I will add a line for it in the docs.

Reply:
Hmm, so does this now pick up the changes if we change the tag? Have you tested it by changing to `0.0.12` and back? It seems weird that the URL approach would work properly while the Git one doesn't.

I'm very against the hash idea. The point of this approach is to make managing dependencies easier than using submodules (i.e. upgrades just require changing a CMake file), whereas with this approach we're adding another thing that has to be updated, and a set of extra commandline steps to generate that thing. This seems like more work than using submodules to me. 

Can you say which part of the docs' recommendation is relevant to us rather than just linking it? AFAICT there are two benefits, security and avoiding unnecessary downloads. The security angle isn't a concern for me, and I can't tell how important the unnecessary downloads bit really is, does it mean without the hash it'll constantly be downloading the .tar again?